File: backend/src/main/java/com/stepx/stepx/dto/OrderShoesDTO.java
package com.stepx.stepx.dto;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

public record OrderShoesDTO(
    Long id,
    LocalDate date,
    String cuponUsed,
    String country,
    String firstName,
    String secondName,
    String email,
    String address,
    String numerPhone,
    BigDecimal summary,
    String state,
    Long userId,
    CouponDTO coupon,
    List<OrderItemDTO> orderItems
) {}

File: frontend/src/app/dtos/ordershoes.dto.ts
import { CouponDTO } from './coupon.dto'; 
import { OrderItemDTO } from './orderitem.dto';  

export interface OrderShoesDTO {
  id: number;
  date: string;     // LocalDate in Java can be represented as an ISO 8601 string in TypeScript (yyyy-mm-dd)
  cuponUsed: string;
  country: string;
  firstName: string;
  secondName: string;
  email: string;
  address: string;
  numerPhone: string;
  summary: number;  
  state: string;
  userId: number;
  coupon: CouponDTO;  
  orderItems: OrderItemDTO[]; 
}
